src/Bazhenov/Task2/MainWindow.cpp

```diff
@@ -0,0 +1,230 @@
+#include "MainWindow.hpp"
+
+#include <vector>
+#include <algorithm>
+
+#include <QOpenGLShader>
+#include <QVector4D>
+
+namespace Bazhenov {
+
+constexpr std::uint32_t STRIDE = 9;
+constexpr std::uint32_t VERTEX_POSITION_OFFSET = 0;
+constexpr std::uint32_t VERTEX_COLOR_OFFSET = 6;
+
+constexpr std::uint32_t VERTEX_MULTIPLICATION_FACTOR = 15;
+
+std::vector<GLfloat> vertices;
+std::vector<GLuint> indices;
+
+constexpr GLfloat VERTICAL_ANGLE = 60.0F;
+constexpr GLfloat NEAR_PLANE = 0.1F;
+constexpr GLfloat FAR_PLANE = 100.0F;
+
+constexpr QVector4D CLEAR_COLOR(0.25F, 0.25F, 0.375F, 1.0F);
+
+void initCube(GLfloat halfWidth, std::uint32_t factor = 1U) {
```

Comment:
Все же не стоит передавать по значению, а затем модифицировать. Я бы сделал `std::abs` и `std::max` в вызывающем коде )

```suggestion
void initCube(const GLfloat halfWidth, const std::uint32_t factor = 1U) {
```

Reply:
Сделано, заменил на Q_ASSERT.